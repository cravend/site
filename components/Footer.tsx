import React, { FC } from 'react';
import {
  Container, Text, Flex,
} from 'theme-ui';
import useLocalization from '../src/hooks/useLocalization';
import Link from './Link';

const Footer: FC = () => {
  const [t] = useLocalization();
  const year = new Date().getFullYear();

  return (
    <Container
      variant="wide"
      as="footer"
      m={0}
      p={0}
      sx={{
        backgroundColor: 'dark',
        height: 100,
        width: '100%',
        display: 'flex',
        alignItems: 'center',
        color: 'white',
      }}
    >
      <Container
        sx={{
          color: 'white',
        }}
      >
        <Flex>
          <Text mr={3}>&copy; {year} {t('name')}</Text>
          <Text>
            {t('viewSource')}{' '}
            <Link to="https://github.com/cravend/site">{t('github')}</Link>
          </Text>
        </Flex>
      </Container>
    </Container>
  );
};

export default Footer;
