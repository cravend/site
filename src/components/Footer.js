import React from 'react'
import styled from 'styled-components'
import { OutboundLink } from 'gatsby-plugin-google-analytics'

const StyledOutboundLink = styled(OutboundLink)`
  position: relative;
  font-weight: 600;
  text-decoration: none;
  color: #000;

  :hover {
    color: #000;
  }

  :before {
    content: '';
    position: absolute;
    width: 100%;
    height: 2px;
    bottom: 0;
    left: 0;
    background-color: #a8dcd9;
    visibility: hidden;
    -webkit-transform: scaleX(0);
    transform: scaleX(0);
    -webkit-transition: all 0.3s ease-in-out 0s;
    transition: all 0.3s ease-in-out 0s;
  }

  :hover:before {
    visibility: visible;
    -webkit-transform: scaleX(1);
    transform: scaleX(1);
  }
`

const First = styled.p`
  margin-bottom: 0;
  @media (min-width: 500px) {
    float: left;
    display: block;
  }
`
const Second = styled.p`
  @media (min-width: 500px) {
    float: right;
    display: block;
    /* float: left; */
  }
`

export default () => (
  <div>
    <First>&copy; 2019 Dalton Craven</First>
    <Second>
      <StyledOutboundLink href="https://github.com/cravend/site">
        View Source on GitHub
      </StyledOutboundLink>
    </Second>
  </div>
)
