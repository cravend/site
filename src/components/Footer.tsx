import { Container, Box, Paragraph, Grid } from "theme-ui";
import useLocalization from "../hooks/useLocalization";
import Link from "./Link";
import type { FC } from "react";

const Footer: FC = () => {
  const [t] = useLocalization();
  const year = new Date().getFullYear();

  return (
    <Box
      as="footer"
      m={0}
      p={0}
      color="white"
      backgroundColor="dark"
      sx={{
        height: 100,
        display: "flex",
        alignItems: "center",
      }}
    >
      <Container>
        <Grid gap={2} columns={"repeat(auto-fit, 200px)"}>
          <Paragraph>{`Â© ${year} ${t("name")} `}</Paragraph>
          <Paragraph>
            {t("view_source")}{" "}
            <Link to="https://github.com/cravend/site">{t("github")}</Link>
          </Paragraph>
        </Grid>
      </Container>
    </Box>
  );
};

export default Footer;
