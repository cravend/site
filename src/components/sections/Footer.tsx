import { useTranslations } from "next-intl";

import styles from "../../styles/modules/sections/Footer.module.scss";
import Link from "../Link";

const Footer = () => {
  const t = useTranslations("sections.Footer");

  return (
    <footer className={styles.background}>
      <div className={styles.container}>
        <p>
          {t.rich("text", {
            date: new Date(),
            Link: (text) => (
              <Link to="https://github.com/cravend/site">{text}</Link>
            ),
          })}
        </p>
      </div>
    </footer>
  );
};

export default Footer;
