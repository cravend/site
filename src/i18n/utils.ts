import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
