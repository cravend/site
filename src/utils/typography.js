import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
