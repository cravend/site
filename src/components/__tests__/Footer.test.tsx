import { render, screen } from "../../tests/utils";
import Footer from "../Footer";

describe("<Footer />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("displays the current year", () => {
    expect.assertions(1);
    render(<Footer />);
    expect(
      screen.getByText(new Date().getFullYear(), { exact: false })
    ).toMatchSnapshot();
  });
});
