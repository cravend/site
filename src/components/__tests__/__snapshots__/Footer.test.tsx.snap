// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> displays the current year 1`] = `
<span
  class="css-1uoge8i"
>
  © 
  2021
   
  Dalton Craven
</span>
`;

exports[`<Footer /> renders 1`] = `
<div>
  <footer
    class="css-1h6jvb9"
  >
    <div
      class="css-xhryvf"
    >
      <div
        class="css-4cffwv"
      >
        <span
          class="css-1uoge8i"
        >
          © 
          2021
           
          Dalton Craven
        </span>
        <span
          class="css-vurnku"
        >
          View source on
           
          <a
            class="css-vurnku"
            href="https://github.com/cravend/site"
          >
            GitHub
          </a>
        </span>
      </div>
    </div>
  </footer>
</div>
`;
